src/test/kotlin/com/exactpro/th2/conn/dirty/fix/TestStrategies.kt

```diff
@@ -0,0 +1,514 @@
+        val handlerSettings = createHandlerSettings(strategyConfig, hbtInt)
+        val channel: IChannel = mock {}
+        val context: IHandlerContext = mock {
+            on { settings }.thenReturn(handlerSettings)
+            on { createChannel(any(), any(), any(), any(), any(), any(), any()) }.thenReturn(channel)
+        }
+        var outgoingSequence = 1
+        val handler = FixHandler(context)
+        val onSendHandler: (ByteBuf, IChannel.SendMode) -> CompletableFuture<MessageID> = { msg, mode ->
+            outgoingSequence++
+            if(enableAdditionalHandling) {
+                val expanded = msg.asExpandable()
+                val metadata = mutableMapOf<String, String>()
+                if(mode == IChannel.SendMode.HANDLE || mode == IChannel.SendMode.HANDLE_AND_MANGLE) {
+                    handler.onOutgoing(channel, expanded, metadata)
+                }
+                sendHandlerExtension(expanded, metadata, mode)
+            }
+            if(msg.contains("35=A\u0001")) {
+                handler.onIncoming(channel, logonResponse(outgoingSequence))
+            }
+            CompletableFuture.completedFuture(MessageID.getDefaultInstance())
+        }
+        var isOpen = false
+        whenever(channel.send(any(), any(), anyOrNull(), any())).doAnswer {
+            onSendHandler(it.arguments[0] as ByteBuf, it.arguments[3] as IChannel.SendMode)
+        }
+        whenever(channel.isOpen).doAnswer {
+            isOpen
+        }
+        whenever(channel.open()).doAnswer {
+            isOpen = true
+            handler.onOpen(channel)
+            mock {  }
+        }
+        whenever(channel.close()).doAnswer {
+            isOpen = false
+            handler.onClose(channel)
+            mock {  }
+        }
+
+        channel.open()
+        clearInvocations(channel)
+
+        return TestContext(channel, handler)
+    }
+
+    companion object {
+        private fun logonResponse(seq: Int): ByteBuf = Unpooled.wrappedBuffer("8=FIXT.1.1\u00019=105\u000135=A\u000134=${seq}\u000149=server\u000156=client\u000150=system\u000152=2014-12-22T10:15:30Z\u000198=0\u0001108=30\u00011137=9\u00011409=0\u000110=203\u0001".toByteArray(Charsets.US_ASCII))
+        private fun businessMessage(seq: Int?, possDup: Boolean = false): ByteBuf = Unpooled.wrappedBuffer("8=FIXT.1.1\u00019=13\u000135=AE${if(seq != null) "\u000134=${seq}" else ""}${if(possDup) "\u000143=Y" else ""}\u0001552=1\u000110=169\u0001".toByteArray(Charsets.US_ASCII))
+        private fun resendRequest(seq: Int, begin: Int, end: Int): ByteBuf = Unpooled.wrappedBuffer("8=FIXT.1.1\u00019=13\u000135=2\u000134=${seq}\u00017=${begin}\u000116=${end}\u0001552=1\u000110=169".toByteArray(Charsets.US_ASCII))
+        private fun testRequest(seq: Int) = Unpooled.wrappedBuffer("8=FIXT.1.1\u00019=13\u000135=1\u000134=${seq}\u0001112=test\u00011552=1\u000110=169".toByteArray(Charsets.US_ASCII))
+    }
```

Comment:
Doesn't the client validate the checksum? Shouldn't we recalculate it when we set a different value as sequence?

Reply:
No, It doesn't. Only updates it in `otgoingUpdateTag`